services/placement/service/placementservice.go

```diff
@@ -84,29 +77,47 @@ func (ps placementService) BuildInitialPlacement(
 		}
 	}
 
-	return p, ps.validateAndSaveSnapshot(p)
+	if err := placement.Validate(p); err != nil {
+		return nil, err
+	}
+
+	// TODO(chaowant) this will be removed once the m3db nodes start to mark shards as available
```

Comment:
I could make init placement to have them as available, I'm fine either way, I just felt that "Initializing" is probably more accurate in describing the state of that shard